src/cache.js

```diff
@@ -231,15 +231,17 @@ class Wrapper {
    */
   async wrapFunction (args, key) {
     const storageKey = this.getStorageKey(key)
-    let data = this.storage.get(storageKey)
-    if (data && typeof data.then === 'function') { data = await data }
+    if (this.ttl > 0) {
+      let data = this.storage.get(storageKey)
+      if (data && typeof data.then === 'function') { data = await data }
 
-    if (data !== undefined) {
-      this.onHit(key)
-      return data
-    }
+      if (data !== undefined) {
+        this.onHit(key)
+        return data
+      }
 
-    this.onMiss(key)
+      this.onMiss(key)
+    }
 
     const result = await this.func(args, key)
 
```

Comment:
L248 returns early when TTL < 1, so `this.storage.set` is never called. Based on this, it seems sensible to bypass the lookup as well, since it's guaranteed not to be there.

Reply:
Unrelated to my proposed change: It seems there's an assumption that TTL cannot have a value in the `(0, 1)` range (excluding 0 and 1), but there is nothing that validates that. Should I change this (L248) check to `if (this.ttl <= 0)` or similar instead?

Reply:
> Unrelated to my proposed change: It seems there's an assumption that TTL cannot have a value in the (0, 1) range (excluding 0 and 1), but there is nothing that validates that. 

why do you say so?

Reply:
(Let me know if I should create a separate issue for this, as I'm unaware of proper PR etiquette for this project :) )

I think the example below illustrates the problem better than my explanation. The TTL is set to 0.9, so the conditional on src/cache.js:L248 evaluates to true and the item isn't stored in the cache. Subsequent lookups result in cache misses. So it seems that in practical terms a TTL `t` where `t < 1` is equivalent to a TTL of 0.

```
$ cat example.mjs 
import { createCache } from 'async-cache-dedupe'

const cache = createCache({
  ttl: 0.9, // seconds
  storage: { type: 'memory' },
})

cache.define('fetchSomething', async (k) => {
  console.log('query', k)

  return { k }
})

const start = Date.now()
await cache.fetchSomething(42)
await cache.fetchSomething(42)
await cache.fetchSomething(42)
console.log('ms', Date.now() - start)
```

```
$ cat package.json 
{
  "dependencies": {
    "async-cache-dedupe": "^1.3.0"
  }
}
```

```
$ node example.mjs 
query 42
query 42
query 42
ms 3
```

If I change the conditional as proposed, the output is:

```
$ node example.mjs 
query 42
ms 5
```

With the change being

```
$ diff -u node_modules/async-cache-dedupe/src/cache.js{.orig,}
--- node_modules/async-cache-dedupe/src/cache.js.orig	2022-06-29 00:25:01.237491670 +0000
+++ node_modules/async-cache-dedupe/src/cache.js	2022-06-29 00:25:18.341747071 +0000
@@ -243,7 +243,7 @@
 
     const result = await this.func(args, key)
 
-    if (this.ttl < 1) {
+    if (this.ttl <= 0) {
       return result
     }
```

I understand that this scenario is not likely to happen, after all, setting such a small TTL is probably an uncommon use case.

Reply:
The assumption is that the number of seconds is an integer, not a float. It would be great to validate that input.

Reply:
I see. I've pushed a new commit that introduces additional validation for the Cache ttl, as well as validation for the function ttl.